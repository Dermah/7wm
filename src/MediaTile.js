import React from "react";
import styled from "styled-components";

const TileBase = styled.div`
  width: 630px;
  height: 354px;
  background-image: url("./tile.jpg");
`;

export default ({ title }) => <TileBase>{title}</TileBase>;
