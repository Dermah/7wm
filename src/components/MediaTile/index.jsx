import * as React from "react";
import styled from "styled-components";

import Channel7Logo from "../Channel7Logo";

const TileBase = styled.div`
  width: 630px;
  height: 354px;
  position: relative;
  margin: 1em;

  background-image: url("${({ src }) => src || "./tile.jpg"}");
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;

  flex-grow: 0
  flex-shrink: 0
`;

const TileTitle = styled.h3`
  font-weight: normal;
  font-size: 1.5em;
  color: white;
  text-align: left;
  background: rgba(0, 0, 0, 0.5);

  width: 100%;
  margin: 0;
  padding: 1em;
  position: absolute;
  bottom: 0;

  display: flex;
  align-items: center;
`;

const TitleLogo = styled(Channel7Logo)`
  margin-right: 0.5em;
`;

export default ({ title, src }) => (
  <TileBase src={src}>
    <TileTitle>
      <TitleLogo />
      {title}
    </TileTitle>
  </TileBase>
);
