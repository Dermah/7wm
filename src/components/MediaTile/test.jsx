import * as React from "react";
import renderer from "react-test-renderer";

import MediaTile from "./";

describe("MediaTile", () => {
  test("snapshot test", () => {
    const tree = renderer.create(<MediaTile />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
