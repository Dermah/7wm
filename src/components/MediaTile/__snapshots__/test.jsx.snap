// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaTile snapshot test 1`] = `
<div
  className="sc-bwzfXH jdYsVk"
  src={undefined}
>
  <h3
    className="sc-htpNat zuFVL"
  >
    <img
      alt="Channel 7 Logo"
      className="sc-bxivhb hzIzrW sc-bdVaJa dGtWke"
      src="logo.png"
    />
  </h3>
</div>
`;

exports[`MediaTile snapshot test with custom image 1`] = `
<div
  className="sc-bwzfXH kDvvKZ"
  src="https://loremflickr.com/320/240/food,cook,kitchen/all"
>
  <h3
    className="sc-htpNat zuFVL"
  >
    <img
      alt="Channel 7 Logo"
      className="sc-bxivhb hzIzrW sc-bdVaJa dGtWke"
      src="logo.png"
    />
  </h3>
</div>
`;
